### DISCUSSION TITLE
All new bugs appear to be pushed directly into the backlog or onto the .NET 7 roadmap, has MS given up?

### DISCUSSION THREAD

**Discussion by JohnHDev** [dotnet/maui#9878] [Q&A]
Im tired of reporting bugs that just don't get fixed, and although .NET MAUI appears to 'mostly' work there is so much that doesn't when you get into the detail.

Why should we raise bugs for this when nothing appears to be getting fixed?
Tons of layout bugs, same if not more than Xamarin Forms, for ex word wrap on iOS not sizing its parent correctly, that's a 5 year old bug in XF, and exists in Maui, Frame doesn't resize correctly (new in Maui), Border doesn't take up all the space its supposed to (new in Maui) and so much more.
Partial methods with return values don't work in multi targeting (bug moved to backlog, this is a feature in C#9!!!)

So I ask, why should we bother?

**Comment by achilleaskar** [comment]
All those years the same things, we support MS at every try, most of the ideas seem awesome but at the end nothing works properly. I remember learning and trying all the new stuff about Windows Phone, spending hours and hours on tutorials and it got canceled. I wish the same thing doesn't happen with MAUI again but as the author says there is an enormous amount of bugs (1.6k at this time) that don't get fixed. Months passing without a release with big changes. It seems to me that MS has 5 people working on MAUI and expects all the rest from the community... And this is sad because we spent so much time supporting those projects. I wish things change faster in the close future as MAUI was supposed to be ready at November 2021 and it will not be at November 2022....

**Reply by jfversluis** [reply]
> but as the author says there is an enormous amount of bugs (1.6k at this time) that don't get fixed.

Those are not all bugs, there is many other things in there. Also, a good amount is fixed each release. Having activity in the issues is sign of an healthy project. People are using it and interacting with it and passionate enough about it to report potential issues. We're very thankful for that. If you look at other open-source projects with a good user base, the number of issues are no different. In my opinion this is in no way a good measurement of how a project is doing.

> and expects all the rest from the community...

We're not expecting anything, but are always very grateful to work together with our wonderful community!

> Months passing without a release with big changes

Since we are now on the .NET schedule, big changes are to be expected each November when a new .NET version lands. That will also be the time when .NET MAUI releases the next major version.

**Comment by jfversluis** [comment]
> has MS given up?

I can assure you that is not the case in any way! If we would have given up, why bother putting things in milestones?

> Why should we raise bugs for this when nothing appears to be getting fixed?

Things being moved to a milestone does not mean it doesn't get fixed. As you might imagine we are working a bit ahead and that is why you might see a lot of things going into .NET 7 as that is coming closer pretty fast.

As for the backlog milestone (and milestones in general), we're busy aligning our way of working to be more in line with the bigger .NET team. In this case we're heavily inspired with how the ASP team is doing things. That way we'll have a more consistent way of working and that should ultimately lead to a clearer overview of what we're working on and why.

More information about the process can be found [here](https://github.com/dotnet/maui/tree/main/docs).

I've also created a small doc with an overview of releases for a quick overview of what has been happening, there you can see that there have been frequent releases and if you click through to them, each release has a good number of bugfixes. It's not been merged at the time of writing, but you can find the PR for it [here](https://github.com/dotnet/maui/pull/9982).

> So I ask, why should we bother?

Each report will make the product better, so thank you for that and please don't stop doing it. We're working hard to make that happen, but as is apparent, we should think about ways to make it more clear!

**Reply by JohnHDev** [reply]
@jfversluis can I ask who made the decision to remove the option of pinning a Maui solution to a specific Maui version and why?
You know how buggy XF has been right? Right?? For years we would never be on the bleeding 'stable' edge, we schedule XF upgrades into our sprints so we can fully regression test. And now MS is saying we must be on the bleeding edge of .NET MAUI! Just as crap as XF, but look new and shiny!

We can hack around this nonsense with rollback files, but why should we support such poor architectural decisions? Ive been in the Xamarin ecosystem since it was MonoTouch, yes I originally paid for this crap. Now Im seriously questioning if we should stay.

I'll be honest here, Im fuming with what MS is doing with Maui.

There is one saving grace for us though, XF support has been extended into 2024. We won't be moving to .NET MAUI any time soon, if ever.

**Comment by janseris** [comment]
> .NET MAUI appears to 'mostly' work there is so much that doesn't when you get into the detail.

This is a precise description of developing in MAUI.

@jfversluis you and @maddymontaquila have an amazing ability to tell that everything is fine when it obviously is not at all.
It's funny when you look at a recent James Montemagno's tutorial and he says something like _"oh I forgot they told me I should not show this control because it's not working"_. How is it **_not working_**? 
It's been already 3 months since official release! Would you put a car model in production if you knew something was not working properly and just waited till everyone crashes and dies and then say thank you for reporting, we'll look into it in 2 years because our team is 3 engineers, 2 managers and 2 youtubers?
 This is what MAUI feels like.
Our company has just finished developing an app in MAUI and what would take us 1 week on desktop took 4 months in MAUI. 
I would not ever believe before using MAUI that a text editor control could be broken in an officially released framework (well, and label (long text, multiline) and button (long text, multiline, disable dstate) and basic layout!)
I've been following MAUI since February 2022 everyday. The situation is terrible. 
To be fair posting the numbers, it's not 1700 issues but it's 1079 bugs. I remember it used to be 300 a few months ago.

>Each report will make the product better,

Yeah but the product should be working now (and it was already supposed to be finished in November 2021) and not in 5 years what is seems with the current tempo (for existing issues only).

What future of MAUI do you expect than when the planning is currently
- .NET 7 + .NET 8: ~380 issues finished by November 2023
- Backlog: 882 issues finished by ??

Quick maths make it look unsustainable.

It's embarassing that app icon isn't still fully working with all its documented properties/settings (Android and Windows).
It's the first thing you see in every app, the influence is 100 % of apps.
I've reported ~10 issues with it in May just when the slogan **_MAUI is Production Ready_** appeared on devblogs.
In the issues, I was just trying the examples from documentation and almost nothing worked besides plain image with no settings and even today, still some persist (I gave up checking if all of the settings are really fixed) and on Android 12, the app icon with no settings is still wrong (I also reported that in May).
It's been 4 months and you cannot fix app icon?  
How can you fix other 100 features of MAUI when you cannot fix 1 feature which has 100 % usage? 

 
Porting bugs from WinUI and Xamarin is another issue. What can we expect when years old bugs from WinUI and Xamarin are ported to MAUI and when reported, we are redirected to Xamarin/WinUI repo where the issue has been open or known for 3 or 4 years without any update (example: photo capture in WinUI and db connection in Xamarin)?

You can't expect much with team with 5 devs which hasn't expanded even after maui got late for. NET 6 back in November 2021. 
Recently got answer with why Visual studio blazor editor is broken from their dev team that the team is small, well **have you tried hiring more devs**? Your employer, Microsoft, is one of the wealthiest corporations on the planet.

https://github.com/dotnet/razor-tooling/issues/6781

>>How is this editor still in such very alpha version (since the launch of new Razor tooling in Visual Studio 2022 based on VS Code editor which was released in late 2021)?

> We are a small, but dedicated, team working to improve things, and we appreciate the issues that you've reported.'

But they will be fixing it in VS 17.4, which looks close to current VS 17.3.3 so it's not just "Backlog" = "never?" as in MAUI (the all time rising number of issues makes it feel that it will be never).

**Reply by achilleaskar** [reply]
> You can't expect much with team with 5 devs which hasn't expanded even after maui got late for. NET 6 back in November 2021. Recently got answer with why Visual studio blazor editor is broken from their dev team that the team is small, well **have you tried hiring more devs**? Your employer, Microsoft, is one of the wealthiest corporations on the planet.

This is exactly what I am trying to understand. I really love you @jfversluis as I  also love @jamesmontemagno. You do an amazing job and your videos help very much but it seems this isn't enough. I created quite a simple app and took me at least ten times the time it would need for WPF and the most sad part was that 90% of the time was about fixing UI bugs in collectionview, swipe view, paddings, buttons shown as deactivated icons in the Shell etc. basic things, nothing special. Yesterday I published my app to take the msix and after saying that manifest is broken tried to fix it and then app wasn't building  because assets.json file had no target. Literally spent 6 hours to fix it! I don't know if problems happen because I started the project when MAUI was beta but MS should hire more people to make MAUI Release ready for real.  

We only complain to you because we loved the idea and want it to work properly because we spent many many hours reporting bugs and reading tutorials to follow up... 

Thanks for your time.

**Reply by mjsb212** [reply]
Microsoft - @jfversluis & @jamesmontemagno - GIVE THESE GUYS RAISES AND BONUSES and / or PROMOTIONS ASAP!!!! - and hire more Devs to fix MAUI! Thank you Gerald and James for the amazing YouTube content and examples, it has helped tremendously and MAUI has INCREDIBLE potential once the major bugs are fixed.
